lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -3998,7 +3998,7 @@ namespace Js
             OUTPUT_VERBOSE_TRACE(Js::DebuggerPhase, _u("DebugProfileProbeThunk: calling function: %s isWrapperRegistered=%d useDebugWrapper=%d\n"),
                 function->GetFunctionInfo()->HasBody() ? function->GetFunctionBody()->GetDisplayName() : _u("built-in/library"), AutoRegisterIgnoreExceptionWrapper::IsRegistered(scriptContext->GetThreadContext()), useDebugWrapper);
 
-            if (scriptContext->IsScriptContextInDebugMode())
+            if (scriptContext->IsScriptContextInDebugRecordingMode())
             {
                 scriptContext->GetDebugContext()->GetProbeContainer()->StartRecordingCall();
```

Comment:
>StartRecordingCall [](start = 71, length = 18)

In ProbeContainer::StartRecordingCall and ProbeContainer::EndRecordingCall can you Assert(this->pScriptContext->GetDebugContext()->IsDebuggerRecording())